docs/AI-for-security/knowledge-base.asciidoc

```diff
@@ -0,0 +1,138 @@
+
+.Requirements
+[sidebar]
+--
+
+* To use Knowledge Base, you need the `Elastic AI Assistant: All` privilege. To edit global Knowledge Base entries (information that will affect the AI Assistant experience for other users in the {kib} space), you need the `Allow Changes to Global Entries` privilege. 
+* You must <<ml-requirements, enable machine learning>> with a minimum ML node size of 4 GB.
+
+--
+
+[discrete]
+[[knowledge-base-rbac]]
+== Role-based access control (RBAC) for Knowledge Base
+
+The `Elastic AI Assistant: All` role privilege allows you to use AI Assistant and access its settings. It has two sub-privileges, `Field Selection and Anonymization`, which allows you to customize which alert fields are sent to AI Assistant and Attack Discovery, and `Knowledge Base`, which allows you to edit and create new Knowledge Base entries.
+
+image::images/knowledge-base-rbac.png[Knowledge base's RBAC settings,60%]
+
+[discrete]
+[[enable-knowledge-base]]
+== Enable Knowledge Base
+
+There are two ways to enable Knowledge Base.
+
+NOTE: You must individually enable Knowledge Base for each {kib} space where you want to use it.
+
+[discrete]
+=== Option 1: Enable Knowledge Base from an AI Assistant conversation
+
+Open a conversation with AI Assistant, select a large language model, then click **Setup Knowledge Base**. If the button doesn't appear, Knowledge Base is already enabled.
+
+image::images/knowledge-base-assistant-setup-button.png[An AI Assistant conversation showing the Setup Knowledge Base button]
+
+Knowledge base setup may take several minutes. It will continue in the background if you close the conversation. After setup is complete, you can access Knowledge Base settings from AI Assistant's conversation settings menu (access the conversation settings menu by clicking the three dots button next to the model selection dropdown).
+
+image::images/knowledge-base-assistant-menu-dropdown.png[AI Assistant's dropdown menu with the Knowledge Base option highlighted]
+
+[discrete]
+=== Option 2: Enable Knowledge Base from the Security AI settings
+
+. To open Security AI settings, use the {kibana-ref}/introduction.html#kibana-navigation-search[global search field] to find "AI Assistant for Security."
+. Click **Setup Knowledge Base**. If the button doesn't appear, Knowledge Base is already enabled.
+
+image::images/knowledge-base-assistant-settings-kb-tab.png[AI Assistant's settings menu open to the Knowledge Base tab]
+
+[discrete]
+[[rag-for-alerts]]
+== Knowledge base for alerts
+When Knowledge Base is enabled, AI Assistant receives alerts from your environment from the last 24 hours which have a status of `open` or `acknowledged`. It uses these as context for each of your prompts. This enables it to answer questions about multiple alerts in your environment rather than just about individual alerts you choose to send it. It receives alerts ordered by risk score, then by the most recently generated. Building block alerts are excluded. 
```

Comment:
```suggestion
When Knowledge Base is enabled, AI Assistant receives `open` or `acknowledged` alerts from your environment from the last 24 hours. It uses these as context for each of your prompts. This enables it to answer questions about multiple alerts in your environment rather than just about individual alerts you choose to send it. It receives alerts ordered by risk score, then by the most recently generated. Building block alerts are excluded. 
```